registry/single_master.go

```diff
@@ -0,0 +1,44 @@
+package registry
+
+import "github.com/mailgun/go-etcd/etcd"
+
+const (
+	masterNodeID = "master"
+)
+
+/*
+SingleMasterRegistry is an implementation of Registry that uses a single master
+instance of an application to handle requests. Internally, this registry uses
+a GroupMasterRegistry with a single, constant group ID.
+*/
+type SingleMasterRegistry struct {
```

Comment:
Don't we want compare and swap in the LB case as well? If we run into some strange scenario where two instances of a service are both trying to register themselves under the same ID, I would rather one accept traffic for one and let the other complain about not being able to register.